src/GDALUtils.cpp

```diff
@@ -100,6 +100,282 @@ ErrorHandler::~ErrorHandler()
+{
+    data.resize(m_raster_x_size * m_raster_y_size);
+
+    GDALRasterBandH band = GDALGetRasterBand(m_ds, nBand);
+    if (!band)
+    {
+        std::ostringstream oss;
+        oss << "Unable to get band " << nBand <<
+            " from data source!";
+        throw pdal_error(oss.str());
+    }
+
+    int nXBlockSize(0);
+    int nYBlockSize(0);
+
+    GDALGetBlockSize(band, &nXBlockSize, &nYBlockSize);
+
+    int nXBlocks = (GDALGetRasterBandXSize(band) + nXBlockSize - 1) / nXBlockSize;
+    int nYBlocks = (GDALGetRasterBandYSize(band) + nYBlockSize - 1) / nYBlockSize;
+
+    for (int iYBlock = 0; iYBlock < nYBlocks; iYBlock++)
+    {
+        int nXValid(0); int nYValid(0);
+        for (int iXBlock = 0; iXBlock < nXBlocks; iXBlock++)
+        {
+
+             if ((iXBlock+1) * nXBlockSize > GDALGetRasterBandXSize(band))
+                 nXValid = GDALGetRasterBandXSize(band) - iXBlock * nXBlockSize;
+             else
+                 nXValid = nXBlockSize;
+             if ((iYBlock+1) * nYBlockSize > GDALGetRasterBandYSize(band))
+                 nYValid = GDALGetRasterBandYSize(band) - iYBlock * nYBlockSize;
+             else
+                 nYValid = nYBlockSize;
+
+            int offset = iXBlock * (nXValid * nYValid) + iYBlock * (nXValid * nYValid);
+
+            CPLErr err = GDALReadBlock(band, iXBlock, iYBlock, data.data() + offset);
+            if (err != CPLE_None)
+            {
+                std::ostringstream oss;
+                oss << "unable to read block for ("<<iXBlock <<","<< iYBlock <<")";
+                throw pdal::pdal_error(oss.str());
+            }
+
+        }
+
+    }
+    return true;
+}
+
+std::vector<std::array<int, 2>> Raster::fetchGDALBlockSizes() const
+{
+    std::vector<std::array<int, 2>> output;
+    for (int i=1; i < m_band_count; ++i)
+    {
+        GDALRasterBandH band = GDALGetRasterBand(m_ds, i);
+        if (!band)
+        {
+            std::ostringstream oss;
+            oss << "Unable to get band " << i <<
+                " from data source!";
+            throw pdal_error(oss.str());
+        }
+        int x(0), y(0);
+        GDALGetBlockSize(band, &x, &y);
+        std::array<int, 2> a;
+        a[0] = x;
+        a[1] = y;
+        output.push_back(a);
+    }
+
+
+    return output;
+}
+
+std::vector<pdal::Dimension::Type::Enum> Raster::computePDALDimensionTypes() const
+{
+
+    if (!m_ds) throw pdal::pdal_error("raster is not open!");
+
+    std::vector<pdal::Dimension::Type::Enum> output;
+    for (int i=0; i < m_band_count; ++i)
+    {
+        GDALRasterBandH band = GDALGetRasterBand(m_ds, i+1);
+        if (!band)
+        {
+            std::ostringstream oss;
+            oss << "Unable to get band " << i+1 <<
+                " from data source!";
+            throw pdal_error(oss.str());
+        }
+
+        GDALDataType t = GDALGetRasterDataType(band);
+        int x(0), y(0);
+        GDALGetBlockSize(band, &x, &y);
```

Comment:
Confused on this one.  What does this do?  It looks like you're fetching x and y and tossing.  All that follows in the PR is stuff about the type.